libnavigation-base/src/main/java/com/mapbox/navigation/base/internal/utils/DirectionsResponseUtils.kt

```diff
@@ -33,7 +34,7 @@ suspend fun parseDirectionsResponse(
         } catch (ex: Exception) {
             when (ex) {
                 is JSONException,
-                is NullPointerException -> ExpectedFactory.createError(ex)
+                is JsonSyntaxException -> ExpectedFactory.createError(ex)
```

Comment:
Now mapbox-java throws a different error:

<img width="1517" alt="image (3)" src="https://user-images.githubusercontent.com/6190346/165731846-507e264a-df06-4996-95ef-429093007122.png">

Reply:
TBH I'm not sure if it's an issue of mapbox-java trowing different exception or us relying on null pointer exception

Reply:
The error type doesn't seem like part of mapbox-java contract. I think we just shouldn't rely on the error type or we need to be ready to update it when mapbox-java changes it.
@mapbox/navigation-android , WDYT?

Reply:
I'm wondering whether we should just catch all exception here and deliver them back to developers as errors.